### PR TITLE
allowing more commands for /note

### PR DESCRIPTION
**Pull request by Herrekt**
Created /note to allow multiple inputs at the same time e.g., after "/note" and "2", can type "search index 5" or "open 2" to immediately search or open the required note.

### CODE REVIEW

File: src/main/java/exceptions/InvalidNoteOpenException.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package exceptions;
+
+public class InvalidNoteOpenException extends InvalidNoteException {
+    public InvalidNoteOpenException(String message) {
+        super(message);
+    }
+}
```
Comment (arcturusz) [inline src/main/java/exceptions/InvalidNoteOpenException.java:7]: Can pull the upstream and do rebase or something cause I think some stuffs are changed? Like now the exceptions are combined to 1 in upstream.

**Review by arcturusz — COMMENTED** [review]
Are you using the latest commit of upstream, cause I think some stuff changed in these files by @peng-217 in the previous PR

File: src/main/java/note/NoteList.java [hunk 2]
```diff
@@ -5,18 +5,21 @@
...
 public class NoteList {
-    private ArrayList<Note> notes;
+    private final ArrayList<Note> notes;
     private final Ui ui;
     private static int defaultTitleCounter = 1;
     private static final String INVALID_NOTE_INDEX_MESSAGE = "The index you entered is not valid!"
             + "Please check again.";
     private static final String INVALID_NOTE_COMMAND_MESSAGE = "The command you entered is not valid!"
-            + "Please check again.";
+            + " Please check again.";
```
Comment (arcturusz) [inline src/main/java/note/NoteList.java:20]: I think can standardize if want a space before the 'Please'

File: src/main/java/note/NoteList.java [hunk 3]
```diff
@@ -124,38 +126,55 @@ public void createNoteProcess(SceneList sceneList) {
...
-        boolean checkExistance = ui.printOpenNoteMessage(this);
-        if (checkExistance) {
+    public void openNoteProcess() throws InvalidNoteException {
+        boolean checkExistence = ui.printOpenNoteMessage(this);
+        if (checkExistence) {
             String userInput = ui.readUserInput();
             while (!userInput.equals("")) {
-                if (!userInput.equals("search") && !userInput.equals("open")) {
+                if (!userInput.startsWith("search") && !userInput.startsWith("open")) {
                     ui.printNoteErrorMessage(INVALID_NOTE_COMMAND_MESSAGE);
```
Comment (arcturusz) [inline src/main/java/note/NoteList.java:135]: I think can do like isValid = !userInput.startsWith("search") && !userInput.startsWith("open") then in the if statement use if(isValid)

File: src/main/java/note/NoteList.java [hunk 4]
```diff
@@ -178,20 +197,58 @@ public void selectSearchMethod(String userInput) throws InvalidNoteException {
...
+    public void indexSearch() throws NumberFormatException {
         ui.printNoteSearchSceneIndexInstructions();
         int sceneIndex = Integer.parseInt(ui.readUserInput());
         ui.printSelectedNote(this.searchNotesUsingSceneIndex(sceneIndex, this));
     }
 
-    public void openNoteDirectly() throws IndexOutOfBoundsException  {
+    public void openNoteDirectly(String index) throws IndexOutOfBoundsException, NumberFormatException {
+        ui.printNoteOpenInstructions();
+        //here the index is not scene index, it is the index in the list
```
Comment (arcturusz) [inline src/main/java/note/NoteList.java:243]: I think our comments follow the format like // Here the index..., with space after // and start with capital letter.

File: src/main/java/note/NoteList.java [hunk 5]
```diff
@@ -124,38 +126,55 @@ public void createNoteProcess(SceneList sceneList) {
         createNote(newNote);
     }
 
-    public void openNoteProcess() {
-        boolean checkExistance = ui.printOpenNoteMessage(this);
-        if (checkExistance) {
+    public void openNoteProcess() throws InvalidNoteException {
+        boolean checkExistence = ui.printOpenNoteMessage(this);
+        if (checkExistence) {
```
Comment (arcturusz) [inline src/main/java/note/NoteList.java:131]: Maybe more boolean sound will be like noteExists?

**Review by peng-217 — APPROVED** [review]
I'll merge it

**Pull request merged**